### PR TITLE
HPITEST-4101 Backend implementation of Timesheet Automation in SSP

### PR DESCRIPTION
**Pull request by nani2196**

### CODE REVIEW

File: src/com/helius/dao/AutomationTimesheetDAOImpl.java
Comment (tirumala-helius) [inline src/com/helius/dao/AutomationTimesheetDAOImpl.java:1]: what is this calcLeaveUsageBasedOnLeaveRecord?
Also there is no session closing in this method

**Review by tirumala-helius — CHANGES_REQUESTED** [review]
what is this calcLeaveUsageBasedOnLeaveRecord?
Also there is no session closing in this method

File: src/com/helius/dao/AutomationTimesheetDAOImpl.java
Comment (nani2196) [reply src/com/helius/dao/AutomationTimesheetDAOImpl.java:1]: this calcLeaveUsageBasedOnLeaveRecord code is from HAP, in leave leave management ->saveOrUpdateEmployeeLeaveData endpoint, for leaveRecordDetails using the same code for updating leave data in BD, Session i will close.

**Comment by tirumala-helius** [comment]
OK. done

File: src/com/helius/dao/AutomationTimesheetDAOImpl.java
Comment (nani2196) [reply src/com/helius/dao/AutomationTimesheetDAOImpl.java:1]: session closing is updated.

**Comment by nani2196** [comment]
Can i Merge this PR, waiting for approval

**Review by tirumala-helius — APPROVED** [review]

**Comment by tirumala-helius** [comment]
approved

**Pull request merged**
